internal/pipeline/executor.go

```diff
@@ -22,174 +22,174 @@ import (
 	"sort"
 	"strings"
 
+	"github.com/GoogleContainerTools/kpt/internal/pkg"
 	"github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1alpha2"
 	"github.com/GoogleContainerTools/kpt/pkg/kptfile"
+	"github.com/GoogleContainerTools/kpt/pkg/kptfile/kptfileutil"
 	"k8s.io/klog"
 	"sigs.k8s.io/kustomize/kyaml/kio"
-	"sigs.k8s.io/kustomize/kyaml/pathutil"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
+// SinkMode determines where the result of hydration be written to.
+type SinkMode string
+
+const (
+	// SinkInplace means that result of hydration will be
+	// written back to the originating source dir.
+	SinkInplace SinkMode = "in-place"
+
+	// SinkOutplace means that result of hydration will be
+	// written to specified dir.
+	SinkOutplace = "out-place"
+
+	// SinkStdout means that the result of hydration will be
+	// written to stdout.
+	SinkStdout = "stdout"
```

Comment:
any reason why SinkInplace is type SinkMode but the other two aren't?

Reply:
Other two also need to have the Type. It was an oversight.
I took out the `sinkMode` functionality for now. `SinkMode` support will be added in a separate PR with `out-of-place` support.